sentry-log/src/integration.rs

```diff
@@ -0,0 +1,161 @@
+
+impl Integration for LogIntegration {
+    fn name(&self) -> &'static str {
+        "log"
+    }
+
+    fn setup(&self, cfg: &mut ClientOptions) {
+        cfg.in_app_exclude.push("log::");
+        cfg.extra_border_frames
+            .push("<sentry_log::Logger as log::Log>::log");
+
+        let filter = self.effective_global_filter();
+        if filter > log::max_level() {
+            log::set_max_level(filter);
+        }
+
+        INIT.call_once(|| log::set_boxed_logger(Box::new(Logger::new())).unwrap());
+    }
+}
+
+impl Default for LogIntegration {
+    fn default() -> Self {
+        Self {
+            global_filter: None,
+            filter: LevelFilter::Info,
+            emit_breadcrumbs: true,
+            emit_error_events: true,
+            emit_warning_events: false,
+            attach_stacktraces: true,
+            dest_log: None,
+        }
+    }
+}
+
+impl LogIntegration {
+    /// Initializes an env logger as destination target.
+    #[cfg(feature = "env_logger")]
+    pub fn with_env_logger_dest(mut self, logger: Option<env_logger::Logger>) -> Self {
+        let logger = logger
+            .unwrap_or_else(|| env_logger::Builder::from_env(env_logger::Env::default()).build());
+        let filter = logger.filter();
+        if self.global_filter.is_none() {
+            self.global_filter = Some(filter);
+        }
+        self.dest_log = Some(Box::new(logger));
+        self
+    }
+
+    /// Returns the effective global filter.
+    ///
+    /// This is what is set for these logger options when the log level
+    /// needs to be set globally.  This is the greater of `global_filter`
+    /// and `filter`.
+    #[inline(always)]
+    pub(crate) fn effective_global_filter(&self) -> LevelFilter {
+        let filter = if let Some(filter) = self.global_filter {
+            if filter < self.filter {
+                self.filter
+            } else {
+                filter
+            }
+        } else {
+            self.filter
+        };
+        std::cmp::max(filter, self.issue_filter())
+    }
+
+    /// Returns the level for which issues should be created.
+    ///
+    /// This is controlled by `emit_error_events` and `emit_warning_events`.
+    #[inline(always)]
+    fn issue_filter(&self) -> LevelFilter {
+        if self.emit_warning_events {
+            LevelFilter::Warn
+        } else if self.emit_error_events {
+            LevelFilter::Error
+        } else {
+            LevelFilter::Off
+        }
+    }
+
+    /// Checks if an issue should be created.
+    pub(crate) fn create_issue_for_record(&self, record: &Record<'_>) -> bool {
+        match record.level() {
+            Level::Warn => self.emit_warning_events,
+            Level::Error => self.emit_error_events,
+            _ => false,
+        }
+    }
+}
+
+pub fn with_integration<F, R>(f: F) -> R
+where
+    F: FnOnce(&LogIntegration) -> R,
+    R: Default,
+{
+    Hub::with_active(|hub| hub.with_integration(f))
```

Comment:
This looks like most integrations will need this -- should we add a shorthand to `Hub` for this?

Reply:
I think I will do that in a followup, yes